src/jsMain/kotlin/net/kyori/adventure/webui/js/Main.kt

```diff
@@ -359,6 +396,49 @@ public fun main() {
         })
 }
 
+private fun readTemplates(): Pair<Map<String, String>, Map<String, String>> {
+    val templatesBox = document.getElementById("templates-box")!!
+    val templateComponents =
+        templatesBox.getElementsByClassName("template-component").asList().map {
+            it.unsafeCast<HTMLInputElement>().checked
+        }
+    val templateKeys =
+        templatesBox.getElementsByClassName("template-key").asList().map {
+            it.unsafeCast<HTMLInputElement>().value
+        }
+    val templateValues =
+        templatesBox.getElementsByClassName("template-value").asList().map {
+            it.unsafeCast<HTMLInputElement>().value
+        }
+    val stringTemplates = mutableMapOf<String, String>()
+    val miniMessageTemplates = mutableMapOf<String, String>()
+    templateKeys
+        .zip(templateValues)
+        .zip(templateComponents)
+        .filter { (t, _) -> t.first.isNotEmpty() && t.second.isNotEmpty() }
+        .forEach { (t, c) ->
+            (if (c) miniMessageTemplates else stringTemplates)[t.first] = t.second
+        }
+    return stringTemplates to miniMessageTemplates
+}
+
+private fun addTemplate(): Triple<HTMLInputElement, HTMLInputElement, HTMLInputElement> {
+    val templatesList = document.getElementById("templates-list")!!
+    lateinit var key: HTMLInputElement
+    lateinit var value: HTMLInputElement
+    lateinit var component: HTMLInputElement
+    templatesList.append {
+        div(classes = "field is-horizontal") {
+            div(classes = "control") {
+                component = input(type = InputType.checkBox, classes = "template-component")
+            }
+            div(classes = "control") { key = input(classes = "template-key") }
+            div(classes = "control") { value = input(classes = "template-value") }
+        }
+    }
+    return Triple(component, key, value)
```

Comment:
Is this return used anywhere?

Reply:
It is used when reading the passed in values in a shared link